### PR TITLE
 Gallery shortcode fixes 

### PR DESCRIPTION
**Pull request by abhishekfdd**
# Gallery shortcode fix

- Added prepare for sql in query method.
- Added Explicit value check for orderby.

### CODE REVIEW

File: app/helper/RTMediaModel.php [hunk 1]
```diff
@@ -122,6 +122,17 @@ public function get( $columns, $offset = false, $per_page = false, $order_by = '
...
+		list( $order_column, $order_direction ) = explode( ' ', $order_by . ' ' ); // Default to space if no direction provided
+
+		if ( ! in_array( strtolower( $order_column ), $allowed_order_columns ) || ! in_array( strtolower( $order_direction ), [
+				'asc',
+				'desc',
+				''
+			] ) ) {
+			$order_by = 'media_id desc'; // Default order
+		}
+
```
Comment (gagan0123) [inline app/helper/RTMediaModel.php:135]:
```
FILE: rtMedia/app/helper/RTMediaModel.php
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 18 ERRORS AFFECTING 8 LINES
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 125 | ERROR | [x] Short array syntax is not allowed (Generic.Arrays.DisallowShortArraySyntax.Found)
 125 | ERROR | [ ] Short array syntax (open) is not supported in PHP 5.3 or lower (PHPCompatibility.Syntax.NewShortArray.Found)
 125 | ERROR | [ ] Short array syntax (close) is not supported in PHP 5.3 or lower (PHPCompatibility.Syntax.NewShortArray.Found)
 125 | ERROR | [ ] Inline comments must end in full-stops, exclamation marks, or question marks (Squiz.Commenting.InlineComment.InvalidEndChar)
 126 | ERROR | [ ] Inline comments must end in full-stops, exclamation marks, or question marks (Squiz.Commenting.InlineComment.InvalidEndChar)
 128 | ERROR | [x] Opening parenthesis of a multi-line function call must be the last content on the line
     |       |     (PEAR.Functions.FunctionCallSignature.ContentAfterOpenBracket)
 128 | ERROR | [x] Only one argument is allowed per line in a multi-line function call (PEAR.Functions.FunctionCallSignature.MultipleArguments)
 128 | ERROR | [x] Short array syntax is not allowed (Generic.Arrays.DisallowShortArraySyntax.Found)
 128 | ERROR | [ ] Short array syntax (open) is not supported in PHP 5.3 or lower (PHPCompatibility.Syntax.NewShortArray.Found)
 129 | ERROR | [x] Array item not aligned correctly; expected 12 spaces but found 16 (WordPress.Arrays.ArrayIndentation.ItemNotAligned)
 130 | ERROR | [x] Array item not aligned correctly; expected 12 spaces but found 16 (WordPress.Arrays.ArrayIndentation.ItemNotAligned)
 131 | ERROR | [x] Array item not aligned correctly; expected 12 spaces but found 16 (WordPress.Arrays.ArrayIndentation.ItemNotAligned)
 131 | ERROR | [x] Each array item in a multi-line array declaration must end in a comma (WordPress.Arrays.CommaAfterArrayItem.NoComma)
 132 | ERROR | [x] Multi-line function call not indented correctly; expected 8 spaces but found 12 (PEAR.Functions.FunctionCallSignature.Indent)
 132 | ERROR | [x] Array closer not aligned correctly; expected 8 space(s) but found 12 (WordPress.Arrays.ArrayIndentation.CloseBraceNotAligned)
 132 | ERROR | [ ] Short array syntax (close) is not supported in PHP 5.3 or lower (PHPCompatibility.Syntax.NewShortArray.Found)
 132 | ERROR | [x] Closing parenthesis of a multi-line function call must be on a line by itself (PEAR.Functions.FunctionCallSignature.CloseBracketLine)
 133 | ERROR | [ ] Inline comments must end in full-stops, exclamation marks, or question marks (Squiz.Commenting.InlineComment.InvalidEndChar)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 11 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

File: app/main/routers/query/RTMediaQuery.php [hunk 2]
```diff
@@ -632,8 +632,8 @@ public function &query( $query ) {
 			$this->query['activity_id'] = array( 'value' );
 			global $wpdb;
 			// todo cache.
-			$sql_query                           = "select id from {$wpdb->prefix}bp_activity where item_id = 0  and type = 'rtmedia_update'";
-			$this->query['activity_id']['value'] = $wpdb->get_col( $sql_query ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+			$sql_query = $wpdb->prepare("SELECT id FROM {$wpdb->prefix}bp_activity WHERE item_id = %d AND type = %s", 0, 'rtmedia_update');
+			$this->query['activity_id']['value'] = $wpdb->get_col( $sql_query );
```
Comment (gagan0123) [inline app/main/routers/query/RTMediaQuery.php:636]:
```
FILE: rtMedia/app/main/routers/query/RTMediaQuery.php
----------------------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------------------------------------
 635 | ERROR | [x] Expected 1 spaces after opening parenthesis; 0 found
     |       |     (PEAR.Functions.FunctionCallSignature.SpaceAfterOpenBracket)
 635 | ERROR | [x] Expected 1 spaces before closing parenthesis; 0 found
     |       |     (PEAR.Functions.FunctionCallSignature.SpaceBeforeCloseBracket)
----------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------------------------------------------------
```

**Review by gagan0123 — CHANGES_REQUESTED** [review]
PR seems good, but let's just correct the PHPCS issues

File: app/helper/RTMediaModel.php (hunk 1 above)
Comment (abhishekfdd) [reply app/helper/RTMediaModel.php:138]: Fixed

File: app/main/routers/query/RTMediaQuery.php (hunk 2 above)
Comment (abhishekfdd) [reply app/main/routers/query/RTMediaQuery.php:636]: Fixed

**Review by gagan0123 — APPROVED** [review]
Looks good

**Pull request merged**